misc/python/materialize/checks/statement_logging.py

```diff
@@ -0,0 +1,56 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.util import MzVersion
+
+
+class StatementLogging(Check):
+    def _can_run(self) -> bool:
+        return self.base_version >= MzVersion(0, 69, 0)
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > ALTER SYSTEM SET statement_logging_max_sample_rate TO 1.0
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > SET statement_logging_sample_rate TO 1.0
+                > SELECT 'hello'; -- Btv was here
+                'hello'
+                > SELECT mz_internal.mz_sleep(20);
```

Comment:
Those sleeps are a bummer -- I recall you had to introduce some in another test as well -- maybe a setting can be implemented that governs the sampling interval , just for the purpose of speeding up the tests?